homeassistant/components/roomba/sensor.py

```diff
@@ -127,10 +132,16 @@ async def async_setup_entry(
     roomba = domain_data.roomba
     blid = domain_data.blid
 
-    async_add_entities(
-        RoombaSensor(roomba, blid, entity_description) for entity_description in SENSORS
+    entities = filter(
+        lambda sensor: sensor.entity_description.exists_fn(sensor),
+        [
+            RoombaSensor(roomba, blid, entity_description)
+            for entity_description in SENSORS
+        ],
     )
```

Comment:
Let's prevent us from even creating these to begin with, instead of filtering them afterward.

Reply:
I've tried that before but in this case I didn't get it to work. That's why I did the filtering instead. I guess I need some help there then.

Reply:
Please, can you tell me how to do this? I couldn't find any examples. The dev docs do not explain how to conditionally add entities.

Reply:
Refactor the exists function to take the `data` and than pass the data so its

`RoombaSensor(roomba, blid, entity_description) for entity_description in SENSORS if entity_description.exists(data)`